wrangler-api/src/main/java/io/cdap/wrangler/api/Directive.java

```diff
@@ -137,4 +143,24 @@ default Relation transform(RelationalTranformContext relationalTranformContext,
     // no-op
     return relation;
   }
+
+  default Optional<ExpressionFactory<String>> getExpressionFactory(RelationalTranformContext ctx) {
```

Comment:
This can be moved to a separate interface as it doesn't make sense for a directive to have this. Maybe a new interface that extends `LinearRelationalTransform` that is internal to Wrangler?